### ISSUE TITLE
LFlare's Writeups

### ISSUE THREAD

**Issue by lflare** [DISMGryphons/GryphonCTF2017-Challenges#1]
Hello,

It has come to my [notice](https://github.com/DISMGryphons/GryphonCTF2017-Challenges/commit/a6888883d38bf25312efe2f3cd8cd382630b0872) that you have removed all references to my **_public_** solutions repository and instead copied over my solutions and writeups to this repository without my knowledge or consent.

Not only was the copy over infringing the **LICENSE** conditions I've clearly included in my repository, some challenges have only been let as `<i>The creator did not published the full solutions for this challenge.</i>`, even though I've clearly published it and linked it from this repository, of which you have removed.

With regards to the **LICENSE** I've included in my repository, you have to follow the [AGPLv3](https://choosealicense.com/licenses/agpl-3.0/) rules and include a copy of the license, for every single solutions/solution you have copied from my repository.

**Comment by zxlim** [comment]
Hi LFlare,

As a Challenge Creator, you are fully obliged, in accordance to the GryphonCTF 2017 Organising Committee ("Committee") guidelines and instructions, to fully publish the challenge solutions in this repository. I have already discussed with you regarding this matter in private. All content created for the purpose of GryphonCTF 2017 shall belong to DISM Gryphons. As some of the provided solution was deemed unacceptable by the Committee, we have redacted the solution to maintain quality control. However they can still be viewed in past commits.

All the solutions present in this repository have been committed previously by their respective challenge creators and no part is reproduced from your writeup linked. They may have been removed prior but nonetheless still remain part of this repository (And thus indirectly DISM Gryphons, as mentioned above). Thus, we have no need to follow the license included in your repository. I have linked the commits below for your reference.

- 8756c10: Pseudoshell solution committed by LFlare in this repository
- 56a3851: 50 Shades Trilogy solution committed by LFlare in this repository

You have the right to create your own writeup, and we will not stop you from doing that. However, linking it in this repository serve no purpose other than self-promotion, and we do not accept that as that is not the point of this repository. You could have uploaded the solutions directly here as required by the Committee, but you did not. The Committee is currently conducting an investigation on your violation.

I will be closing this issue. Feel free to reopen this issue if I have not completely resolved your queries.

**Comment by lflare** [comment]
> As a Challenge Creator, you are fully obliged, in accordance to the GryphonCTF 2017 Organising Committee ("Committee") guidelines and instructions, to fully publish the challenge solutions in this repository.

No I am not _obliged_ to do so, I am a volunteering challenge creator who is under no specifically declared agreements/contracts/clauses/disclaimers to do any special "obligations".

Additionally, I have even informed @Yong-Ze during the week of the competition and informed him that I would not be committing my solutions to the repository and he did not have an issue with it. However, you, @zxlim, of whom I believe is not part of Gryphon 17/18, do. Pray tell, why the fuss about it?

    [2:55 PM] void: Btw will u upload your solutions to the repo?
    [2:55 PM] void: Ohh nice haha
    [2:56 PM] LFlare: Actually, I was thinking of making a writeup for all the challenges in another repo
    [2:56 PM] LFlare: will upload it but not in the same repo
    [2:56 PM] void: Sounds good
    [2:56 PM] void: Share that repo with me so I cn put it on the Gryphons repo
    [2:57 PM] void: Like the one under DISM Gryphons(edited)
    [2:57 PM] LFlare: feel free to fork it after the ctf(edited)
    [2:57 PM] void: Sure haha

> All content created for the purpose of GryphonCTF 2017 shall belong to DISM Gryphons.

Unless expressly declared, or under the clause of an agreement, code ownership defaults under the author. If you did not get the memo, I did not sign or agree to give up the full ownership of anything I have created or coded for GryphonCTF 2017. In fact, no such agreement was even brought up to my attention. The only disclaimer I have ever submitted was for the first installment of "50 Shades of Pixels", to expressly grant the rights of which to "DISM Gryphons".

> All the solutions present in this repository have been committed previously by their respective challenge creators and no part is reproduced from your writeup linked. They may have been removed prior but nonetheless still remain part of this repository (And thus indirectly DISM Gryphons, as mentioned above). Thus, we have no need to follow the license included in your repository.

Like previously mentioned, my solutions do not belong to "DISM Gryphons". You do not have the rights to commit it back under your own name taking up ownership or otherwise.

> However, linking it in this repository serve no purpose other than self-promotion, and we do not accept that as that is not the point of this repository.

Incorrect, linking it here does not serve myself any benefits. You gave me a deadline of which after the deadline has passed, I'm unable to edit my writeups any longer. This is why I've linked it in the first place is so that I would have continous editorial access to my own writeups to be able to provide up-to-date writeups to readers. It is for an educational purpose.

> You could have uploaded the solutions directly here as required by the Committee, but you did not.

Read the entire message I've posted. No one had any issues when I did not upload solutions to my challenges, not even @Yong-Ze. You all used my challenges during the CTF and had no issues with the lack of solutions in the repository, even after I had told @Yong-Ze I would not be pushing solutions to the repository.

> The Committee is currently conducting an investigation on your violation.

Oh? Violation? I'll be more happy to entertain any proofs of agreements/contracts/clauses/disclaimers that I've signed/agreed to.

TLDR: Since you did not want to settle this the easy way and include a simple AGPLv3 license file, I am now hereby requesting for full removal of all my written code, owned by me. This includes all but "50-Shades-of-Pixels".

_SIDE-NOTE: Considering the recent edits to the CTF framework, particularly over Genesis, you are more than welcomed to remove Genesis 6, as the disclaimers you have placed has completely ruined that challenge for anyone ever wanting to attempt it._

**Comment by zxlim** [comment]
Hi LFlare,

For your information, I am currently an Advisor for the current DISM Gryphons Executive Committee. Thus I am allowed to represent or give instructions on behalf of DISM Gryphons. If you have doubts regarding this statement, you may approach @Yong-Ze to confirm this.

Reasons for adding the disclaimer in the Genesis README was for indemnity and legal purposes.

After careful consideration and seeking advice from a legal perspective, we have decided to not further complicate this issue and remove all your challenges from this repository. Please see commit 287cb4a. With immediate effect, the Executive Committee has officially withdrawn recognition of your appointment as _Event Helper (Challenge Creator)_ for the event "GryphonCTF 2017".

On behalf of the GryphonCTF 2017 Organising Committee, I thank you for your prior contributions and services rendered.

**Comment by bitxer** [comment]
Hi LFlare,

After discussions with the GryphonCTF 2017 Organising Committee and you, we have decided to include the AGPL-3.0 licence for the solutions to challenges that was created by you. However, the license for the rest of the repository, that is not included in the solutions folder of his challenges, shall remain as GPL-3.0.

In addition, with immediate effect, the Executive Committee has officially reinstated recognition for your appointment as Event Helper (Challenge Creator) for the event "GryphonCTF 2017".
